fc/parsing/actions.py

```diff
@@ -882,6 +876,178 @@ def _expr(self):
         self.rule = self.tokens[-1]
 
 
+class ProtocolVariable():
+    """
+    Collates information about a (new or existing) model variable used in a protocol.
+
+    Properties:
+
+    ``name``
+        A prefixed ontology term or a local variable name.
+    ``short_name``
+        A short name (no prefixes) that can be used for this variable.
+    ``long_name``
+        This variable's name plus any known aliases.
+    ``input_terms``
+        A list of rdf terms by which this variable is known as an input. This is typically zero or one term, but could
+        end up being more if two ontology terms specify the same model variables. For inputs this is an error.
```

Comment:
Actually, I don't think this does _need_ to be an error for inputs either, provided units are consistent?

Reply:
Yes, it would get confusing! I'm happy to make it an error.